From d6abe5316146892c78bf0678598f3b71ebcce5d4 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:29:23 -0400
Subject: [alsa] ice1712: Add S/PDIF suspend support for ICE1712-based M-Audio cards

Message-id: <140972823573-253-patchset-jkysela@redhat.com>
Patchwork-id: 90953
O-Subject: [RHEL 7.1 PATCH 252/442] ALSA: ice1712: Add S/PDIF suspend support for ICE1712-based M-Audio cards
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Add S/PDIF suspend support for M-Audio cards based on ICE1712 chip.
Tested (playback only) on Audiophile 24/96. Capture will probably not work.

Signed-off-by: Ondrej Zary <linux@rainbow-software.org>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: d272ccd0d084766a3e2adba10bea47091b6d68e7)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/ice1712/delta.c b/sound/pci/ice1712/delta.c
index ed2144e..b3eed8d 100644
--- a/sound/pci/ice1712/delta.c
+++ b/sound/pci/ice1712/delta.c
@@ -580,6 +580,28 @@ static struct snd_ak4xxx_private akm_vx442_priv = {
 static int snd_ice1712_delta_resume(struct snd_ice1712 *ice)
 {
  unsigned char akm_backup[AK4XXX_IMAGE_SIZE];
+
+ /* init spdif */
+ switch (ice->eeprom.subvendor) {
+ case ICE1712_SUBDEVICE_AUDIOPHILE:
+ case ICE1712_SUBDEVICE_DELTA410:
+ case ICE1712_SUBDEVICE_DELTA1010E:
+ case ICE1712_SUBDEVICE_DELTA1010LT:
+ case ICE1712_SUBDEVICE_VX442:
+ case ICE1712_SUBDEVICE_DELTA66E:
+  snd_cs8427_init(ice->i2c, ice->cs8427);
+  break;
+ case ICE1712_SUBDEVICE_DELTA1010:
+ case ICE1712_SUBDEVICE_MEDIASTATION:
+  /* nothing */
+  break;
+ case ICE1712_SUBDEVICE_DELTADIO2496:
+ case ICE1712_SUBDEVICE_DELTA66:
+  /* Set spdif defaults */
+  snd_ice1712_delta_cs8403_spdif_write(ice, ice->spdif.cs8403_bits);
+  break;
+ }
+
  /* init codec and restore registers */
  if (ice->akm_codecs) {
   memcpy(akm_backup, ice->akm->images, sizeof(akm_backup));
-- 
1.7.1